common/workunit/workunit.cpp

```diff
@@ -6336,10 +6341,60 @@ IStringVal& CLocalWorkUnit::getSecurityToken(IStringVal &str) const
     return str;
 }
 
-void CLocalWorkUnit::setSecurityToken(const char *value)
+void CLocalWorkUnit::getDistributedAccessToken(IStringVal & datoken) const
 {
     CriticalBlock block(crit);
-    p->setProp("@token", value);
+    datoken.set(p->queryProp("@distributedAccessToken"));
+}
+
+void CLocalWorkUnit::setDistributedAccessToken(const char * datoken)
+{
+    CriticalBlock block(crit);
+//TODO This method probably isn't needed. If so, should I verify it here?
+    p->setProp("@distributedAccessToken", datoken);
+}
+
+bool CLocalWorkUnit::setDistributedAccessToken(const char * wuid, const char * user)
+{
+    CriticalBlock block(crit);
+    if (isEmptyString(user))
+    {
+        ERRLOG("Cannot create workunit Distributed Access Token, user not present");
+        return false;
+    }
+
+    if (isEmptyString(wuid))
+    {
+        ERRLOG("Cannot create workunit Distributed Access Token, wuid ID not present");
+        return false;
+    }
+
+//TODO Should I verify the given wuid/user match what's in the wu ?
```

Comment:
They should definitely match.  I would even say just use the owner, rather than passing in something else.

Reply:
Still haven't looked at how these things are called enough to know if the encapsulation of who can create the token and when is correct.  But in any case the user should be the owner.